### DISCUSSION TITLE
Frequency map with different coordinates

### DISCUSSION THREAD

**Discussion by TeresiaOlsson** [atcollab/at#755] [Python AT]
Hi @oscarxblanco 

I saw that you have implemented the ``fmap_parallel_track`` function. Do you already have plans to expand it to more coordinates? For example, in addition to calculating in x and y I would like to be able to calculate in x and delta.

I think it would be nice if the planes can be chosen in the same way as for ``get_acceptance``.

If you are not already working on it, I might make an attempt to modify the function to include this.

**Comment by oscarxblanco** [comment]
Dear Teresia,
that seems to me quite useful. I have not done it yet. Please proceed if you have the time.
Remember that the frequency analysis is now parallelized https://github.com/atcollab/at/pull/681 (work done by @swhite2401).

On top of that, I had the idea that one could give the chance to the user to pass a custom particle array, i.e.  one constructs the particle array as needed and the function tracks and get the harmonic components. In that way you would be able to get the tune shifts wrt to any parameter.

At the moment I'm working on momentum aperture and physical apertures on pyat and AT matlab, and it will take me a while to check all results. I think I will be able to work on the frequency map after that if you decide to postpone it.

Best regards,
o

**Comment by oscarxblanco** [comment]
Dear Teresia,
last night I had time to do some modifications on the frequency analysis, and I have a working function already that needs to be checked. Unfortunately, I won't be able to dedicate more time this week  to this, but, feel free to try it out.

Here is the link to the branch, https://github.com/atcollab/at/pull/756

It gives me the same result as the older frequency map analysis when using ['x','y'] coordinates. It also runs with  ['x','dp'] but I wouldn't know if the result is correct. Please, check it carefully.

Best regards,
o
